pkg/integrations/v4/logs/cfg.go

```diff
@@ -344,6 +383,85 @@ func parseTcpInput(l LogCfg) (input FBCfgInput, filters []FBCfgParser, err error
 	return input, filters, nil
 }
 
+//Winlog: "winlog" plugin
+func parseWinlogInput(l LogCfg, dbPath string) (input FBCfgInput, filters []FBCfgParser, err error) {
+	input = newWinlogInput(*l.Winlog, dbPath, l.Name)
+	filters = append(filters, newRecordModifierFilterForInput(l.Name, fbInputTypeWinlog, l.Attributes))
+	scriptContent, err := createLuaScript(*l.Winlog)
+	if err != nil {
+		return FBCfgInput{}, []FBCfgParser{}, err
+	}
+	scriptName, err := saveToTempFile([]byte(scriptContent))
+	if err != nil {
+		return FBCfgInput{}, []FBCfgParser{}, err
+	}
+	eventIdLuaFilter := newLuaFilter(l.Name, scriptName)
+	filters = append(filters, eventIdLuaFilter)
+
+	return input, filters, nil
+}
+
+func createLuaScript(winlog LogWinlogCfg) (scriptContent string, err error) {
+	var fbLuaScript FBWinlogLuaScript
+	fbLuaScript.FnName = fbLuaFnNameWinlogEventFilter
+	included, excluded := winlog.CollectEventIds, winlog.ExcludeEventIds
+	fbLuaScript.IncludedEventIds, err = createConditions(included, "true")
+	if err != nil {
+		return "", err
+	}
+	fbLuaScript.ExcludedEventIds, err = createConditions(excluded, "false")
+	if err != nil {
+		return "", err
+	}
+	return fbLuaScript.Format()
+}
+
+func createConditions(numberRanges []string, defaultIfEmpty string) (conditions string, err error) {
+	if len(numberRanges) > 0 {
+		conditions := make([]string, 0, len(numberRanges))
+		for _, numberRange := range numberRanges {
+			if match, err := regexp.MatchString(eventIdRangeRegex, numberRange); match && err == nil {
+				//EventID range in the format 1234-2345
+				var splitRange = strings.Split(numberRange, "-")
+				bottomLimit, _ := strconv.Atoi(splitRange[0])
+				topLimit, _ := strconv.Atoi(splitRange[1])
+				if bottomLimit > topLimit {
+					topLimit, bottomLimit = bottomLimit, topLimit
+				}
+				conditions = append(conditions, fmt.Sprintf("eventId>=%d and eventId<=%d", bottomLimit, topLimit))
+			} else if _, err := strconv.Atoi(numberRange); err == nil {
+				//Single EventID
+				conditions = append(conditions, fmt.Sprintf("eventId==%s", numberRange))
+			} else {
+				//Invalid format
+				return "", fmt.Errorf("winlog: invalid range format or number")
+			}
+		}
+
+		return strings.Join(conditions, " or "), nil
+	} else {
+
+		return defaultIfEmpty, nil
+	}
+}
+
+func saveToTempFile(config []byte) (string, error) {
+	// create it
+	file, err := ioutil.TempFile("", "nr_fb_lua_filter")
```

Comment:
Hi, could you clear up the temp files created after the tests are run. An example of this can be found here https://github.com/newrelic/infrastructure-agent/blob/a6500b749ce46fe55ebdfbb8db9edab1a5347830/pkg/integrations/v4/manager_test.go#L142

Reply:
Agreed!  creates a single temp file, now I added a function that removes it after executing the test.

Reply:
Thank you 👍